zerver/views/custom_profile_fields.py

```diff
@@ -116,20 +116,48 @@ def validate_custom_profile_field(
     validate_display_in_profile_summary_field(field_type, display_in_profile_summary)
 
 
+def validate_custom_profile_field_update(
+    field_type: int,
+    field_data: ProfileFieldData,
+    display_in_profile_summary: bool,
+    name: Optional[str] = None,
+    hint: Optional[str] = None,
+) -> None:
+    # Validate field data
+    validate_custom_field_data(field_type, field_data)
+
+    if not is_default_external_field(field_type, field_data):
+        # If field is default external field then we will fetch all data
+        # from our default field dictionary, so no need to validate name or hint
+        # Validate field name, hint if not default external account field
+        try:
+            if hint:
+                hint_validator("hint", hint)
+            if name is not None:
+                if not name.strip():
+                    raise JsonableError(_("Label cannot be blank."))
+                else:
+                    name_validator("name", name)
+        except ValidationError as error:
+            raise JsonableError(error.message)
+
```

Comment:
If you compare to `validate_custom_profile_field`, you'll notice that we don't check for invalid field types. This is because we don't allow to update the field type and therefore no validation needed.

Reply:
Does it make sense to record that idea in a comment?

Reply:
Yeah, good idea, thanks! In that case, I've made changes not to copy `validate_custom_field_data` but instead call it after populating relevant fields from the db. This might have some unnecessary checks like field_type (it cannot be updated), but it will be much more maintainable in the long run